### PR TITLE
Enable faster retrieval times from Box provider

### PR DESCRIPTION
**Pull request by awead**
* adds unit tests for BE::Driver::Box
* retrieving data by requesting unique ids for files and folders instead of using paths
* store token expiration time in session
* refactor code for clarity

### CODE REVIEW

**Comment by awead** [comment]
ping @cam156

File: lib/browse_everything/driver/box.rb [hunk 1]
```diff
@@ -10,103 +11,107 @@ def icon
...
-            access_token: token
-          )
-          result = new_session.get("#{RubyBox::API_URL}/users/me")
-          result['status'] != 200
-        rescue RubyBox::AuthError => e
-          Rails.logger.error("AuthError occured when checking token. Exception #{e.class.name} : #{e.message}. token as expired and need to refresh it")
-          return true
+        # @param [OAuth2::AccessToken] access_token
+        def register_access_token(access_token)
+          @token = {
```
Comment (cam156) [inline lib/browse_everything/driver/box.rb]: @awead is there a reason you are not just toring the accesss_token and making a hash here?

File: spec/unit/box_spec.rb [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+  let(:browser) { BrowseEverything::Browser.new(url_options) }
+  let(:provider) { browser.providers['box'] }
+  let(:auth_params) do
+    {
+      'code' => 'CODE',
+      'state' => 'box'
+    }
+  end
+
+  # To re-record the VCR responses, you will need to substitute these values with the actual tokens returned from Box.
```
Comment (cam156) [inline spec/unit/box_spec.rb]: 👏   Did you write up some instructions on recording the vcr repsonses?

File: spec/unit/box_spec.rb [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+  end
+
+  describe '#contents' do
+    before { provider.token = token }
+
+    context 'of the root directory' do
+      let(:contents) { provider.contents('') }
+
+      describe 'the first item' do
+        subject { contents[0] }
```
Comment (cam156) [inline spec/unit/box_spec.rb]: This contents [0] is a bit magical.  I wonder if a comment here to point to the VCR tape may make it less so.

**Comment by awead** [comment]
@cam156 I've added some comments and restructured the test a bit, which hopefully makes it clearer.

**Pull request merged**